gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/GobblinInputFormat.java

```diff
@@ -0,0 +1,156 @@
+ */
+
+package gobblin.runtime.mapreduce;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+
+import com.google.common.collect.Lists;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * An input format for reading Gobblin inputs (work unit and multi work unit files).
+ */
+public class GobblinInputFormat extends InputFormat<LongWritable, Text> {
+
+  @Override
+  public List<InputSplit> getSplits(JobContext context)
+      throws IOException, InterruptedException {
+
+    Path[] inputPaths = FileInputFormat.getInputPaths(context);
+    if (inputPaths == null || inputPaths.length == 0) {
+      throw new IOException("No input found!");
+    }
+
+    List<InputSplit> splits = Lists.newArrayList();
+
+    for (Path path : inputPaths) {
+      FileSystem fs = path.getFileSystem(context.getConfiguration());
+      FileStatus[] inputs = fs.listStatus(path);
+
+      if (inputs == null) {
+        throw new IOException(String.format("Path %s does not exist.", path));
+      }
+      for (FileStatus input : inputs) {
+        splits.add(new GobblinSplit(input.getPath().toString()));
+      }
+    }
+    return splits;
+  }
+
+  @Override
+  public RecordReader<LongWritable, Text> createRecordReader(InputSplit split, TaskAttemptContext context)
+      throws IOException, InterruptedException {
+    return new GobblinRecordReader((GobblinSplit) split);
+  }
+
+  /**
+   * {@link InputSplit} that just contain the work unit / multi work unit file that each mapper should process.
+   */
+  @AllArgsConstructor
+  @NoArgsConstructor
+  public static class GobblinSplit extends InputSplit implements Writable {
+
+    @Getter
+    private String path;
+
+    @Override
+    public void write(DataOutput out)
+        throws IOException {
+      out.writeUTF(this.path);
+    }
+
+    @Override
+    public void readFields(DataInput in)
+        throws IOException {
+      this.path = in.readUTF();
+    }
+
+    @Override
+    public long getLength()
+        throws IOException, InterruptedException {
+      return 0;
+    }
+
+    @Override
+    public String[] getLocations()
+        throws IOException, InterruptedException {
+      return new String[0];
+    }
+  }
+
+  /**
+   * Returns a single record containing the name of the work unit / multi work unit file to process.
```

Comment:
As a side not, I think we should later change the input format to actually return the work units. This will allow us to store work units in something like HBase or some such.